airflow/upgrade/rules/no_additional_args_in_operators.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import absolute_import
+
+import os
+import re
+import logging
+import warnings
+from airflow.utils.dag_processing import correct_maybe_zipped, list_py_file_paths
+from airflow import conf
+from airflow.models.dagbag import DagBag
+from airflow.upgrade.rules.base_rule import BaseRule
+
+
+class NoAdditionalArgsInOperatorsRule(BaseRule):
+    title = "No additional argument allowed in BaseOperator."
+
+    description = """
+                Passing unrecognized arguments to operators is not allowed in Airflow 2.0 anymore,
+                and will cause an exception.
+                  """
+
+    def check(self, dags_folder=None):
+        if not dags_folder:
+            dags_folder = conf.get("core", "dags_folder")
```

Comment:
```suggestion
    def check(self):
        dags_folder = conf.get("core", "dags_folder")
```
I don't think users will be able to specify the dag folder in other way

Reply:
ok.

Reply:
This is needed for testing.

Reply:
This is a similar approach that was taken in the https://github.com/apache/airflow/blob/f47efe91a8b4ba5ba43fa8d55990d27516ab0b6c/airflow/upgrade/rules/undefined_jinja_varaibles.py#L133